pkg/controller/backup/backup_controller.go

```diff
@@ -197,6 +203,42 @@ func (c *Controller) updateBackup(cur interface{}) {
 	c.enqueueBackup(newBackup)
 }
 
+func (c *Controller) deleteJob(obj interface{}) {
+	job, ok := obj.(*batchv1.Job)
+	if !ok {
+		return
+	}
+
+	ns := job.GetNamespace()
+	jobName := job.GetName()
+	backup := c.resolveBackupFromJob(ns, job)
+	if backup == nil {
+		return
+	}
+	klog.V(4).Infof("Job %s/%s deleted through %v.", ns, jobName, utilruntime.GetCaller())
+	c.updateBackup(backup)
+}
+
+func (c *Controller) resolveBackupFromJob(namespace string, job *batchv1.Job) *v1alpha1.Backup {
+	owner := metav1.GetControllerOf(job)
+	if owner == nil {
+		return nil
+	}
+
+	if owner.Kind != controller.BackupControllerKind.Kind {
+		return nil
+	}
+
+	backup, err := c.deps.BackupLister.Backups(namespace).Get(owner.Name)
```

Comment:
do we need to handle error?

Reply:
I copy code from https://github.com/pingcap/tidb-operator/blob/0dc4c4f3d878529a7c49f64e99cd992d55236bc9/pkg/controller/tidbcluster/tidb_cluster_controller.go#L250

Or, we add some logs to print errors?